homeassistant/components/google/__init__.py

```diff
@@ -250,31 +249,30 @@ async def async_setup_services(
 ) -> None:
     """Set up the service listeners."""
 
-    created_calendars = set()
     calendars = await hass.async_add_executor_job(
         load_config, hass.config.path(YAML_DEVICES)
     )
 
-    async def _found_calendar(call: ServiceCall) -> None:
-        calendar = get_calendar_info(hass, call.data)
-        calendar_id = calendar[CONF_CAL_ID]
-
-        if calendar_id in created_calendars:
-            return
-        created_calendars.add(calendar_id)
-
+    async def _found_calendar(calendar_item: Calendar) -> None:
+        calendar = get_calendar_info(
+            hass,
+            {
+                **calendar_item.dict(exclude_unset=True),
+                CONF_TRACK: track_new,
+            },
+        )
         # Populate the yaml file with all discovered calendars
-        if calendar_id not in calendars:
-            calendars[calendar_id] = calendar
+        if calendar_item.id not in calendars:
```

Comment:
I'd store the calendar id in a local variable since we'll use it more than once here. We'll save some attribute access. I don't think it matters here but it's a good habit.

Reply:
👍🏼